examples/07-routing/README.md

```diff
@@ -2,12 +2,62 @@
 
 # 07 Routing
 
-We haven't quite got round to documenting this example yet. You can still check out the [source code](./src/app.gleam) and run it with the following command:
+In this example, we demonstrate basic routing using the community library [modem](https://hexdocs.pm/modem/). Modem's quickstart docs should be all you should need to get up to speed, so that's the best place to start.
 
-```sh
-gleam run -m lustre/dev start --use-example-styles
+Of course, it's not much fun routing without something to route _to_. This example lets users create new pages on the fly - a guest book for your next house party! Hospitality is very important, and guests will be sure to feel welcome when they see their name in the navigation with a special greeting page just for them.
+
+## Using Modem
+
+Modem uses [custom side effects](../06-custom-effects/) and external Javascript to translate browser click and navigation events into `update` messages for the Lustre runtime. All we need to use it is a [route change handler function](./src/app.gleam#L59) that we can pass to `modem.init` in our app's `init` function.
+
+> _Note:_ See [`modem.advanced`](https://hexdocs.pm/modem/modem.html#advanced) to configure more options.
+
+Inside our `on_route_change` function, we match URI path segment patterns to our routes:
+
+```rust
+  fn on_route_change(uri: Uri) -> Msg {
+    case uri.path_segments(uri.path) {
+      ["welcome", guest] -> OnRouteChange(WelcomeGuest(guest))
+      _ -> OnRouteChange(Home)
+    }
+  }
+```
+
+In our `update` function, we assign the matched route to `model.current_route`:
+
+```rust
+fn update(model: Model, msg: Msg) -> #(Model, Effect(Msg)) {
+  case msg {
+    OnRouteChange(route) -> #(
+      Model(..model, current_route: route),
+      effect.none(),
+    )
+  ...
+  }
 ```
 
+And in our `view` function, we use `model.current_route` to determine what to display to the user:
+
+```rust
+fn view(model: Model) -> Element(Msg) {
+  let page = case model.current_route {
+    Home -> render_home(model)
+    WelcomeGuest(name) -> render_welcome(model, name)
+  }
+  ...
+}
+```
+
+## Views: They're just functions!
```

Comment:
I like this section but I don't think it's landing quite right. I think the thing to emphasise here is because view functions are just functions, we can use normal Gleam things like _pattern matching on the route_ to choose what to render.

Reply:
I took another crack at this! Lemme know how that sits with you!